nncf/torch/tensor_statistics/collectors.py

```diff
@@ -71,11 +71,14 @@ def max(cls, *args) -> NNCFTensor:
         return cls.reduce_max(stacked, axis=0, keepdims=False)
 
     @staticmethod
-    def mean(x: NNCFTensor, axis: Union[int, Tuple[int, ...], List[int]], keepdims=False) -> NNCFTensor:
-        return PTNNCFTensor(x.tensor.mean(dim=axis, keepdim=keepdims))
+    def mean(x: NNCFTensor, axis: Union[int, Tuple[int, ...], List[int]], keepdims: bool = False) -> NNCFTensor:
+        comp_dtype, out_dtype = _get_computing_dtype(x.tensor.dtype)
+        return PTNNCFTensor(
+            torch.mean(x.tensor, axis=axis, keepdims=keepdims, dtype=comp_dtype).to(dtype=out_dtype, copy=False)
+        )
 
     @staticmethod
-    def median(x: NNCFTensor, axis: Union[int, Tuple[int, ...], List[int]], keepdims=False) -> NNCFTensor:
+    def median(x: NNCFTensor, axis: Union[int, Tuple[int, ...], List[int]], keepdims: bool = False) -> NNCFTensor:
         # See https://github.com/pytorch/pytorch/issues/61582
```

Comment:
`_get_computing_dtype` is missed for the median function

Reply:
Done. For quantile it is not needed.